web-ui/src/main/resources/catalog/components/doi/DoiService.js

```diff
@@ -85,24 +108,55 @@
        *     the metadata is published.
        *
        */
-      function canPublishDoiForResource(md, resource) {
+      function canPublishDoiForResource(md, doiUrl) {
+        if (doiUrl == null || doiUrl.indexOf("doi.org/") === -1) {
+          return false;
+        }
+
         var doiKey = gnConfig["system.publication.doi.doikey"];
         var isMdWorkflowEnableForMetadata =
           gnConfig["metadata.workflow.enable"] && md.draft === "y";
         return (
           isDoiApplicableForMetadata(md) &&
-          resource.lUrl !== null &&
-          resource.lUrl.match("doi.org/" + doiKey) !== null &&
+          doiUrl.match("doi.org/" + doiKey) !== null &&
           !isMdWorkflowEnableForMetadata
         );
       }
 
+      function checkDoiManagementForResource(md, resource) {
+        if (resource.locUrl == null || resource.locUrl.indexOf("doi.org/") === -1) {
+          return;
+        }
+        var doiKey = null;
+        getDoiServersForMetadata(md.id).then(function (response) {
+          var isMdWorkflowEnableForMetadata =
+            gnConfig["metadata.workflow.enable"] && md.draft === "y";
+          for (var i = 0; i < response.data.length; i++) {
+            if (resource.locUrl.match("doi.org/" + response.data[i].prefix)) {
+              doiKey = response.data[i].prefix;
+              break;
+            }
+          }
+          if (doiKey !== null) {
+            if (
+              isDoiApplicableForMetadata(md) &&
+              resource.locUrl.match("doi.org/" + doiKey) !== null &&
```

Comment:
This check seems a bit redundant with the check done in https://github.com/geonetwork/core-geonetwork/pull/8468/files#diff-031877af43ecd9313d60230ea5001dac1160701606a4fea6c33b0cd15d476beaR135, no?